lib/manageiq/messaging/kafka/client.rb

```diff
@@ -69,6 +69,11 @@ def topics
           Rdkafka::Metadata.new(native_kafka).topics.collect { |topic| topic[:topic_name] }
         end
 
+        def metadata
```

Comment:
I think "metadata" is a little vague and would be hard to implement for the other messaging types, where listing queues/topics could likely be done.

I think we'd be better off fixing the "topics" method above, possibly implement an `alive?` method that catches `Local: Broker transport failure (transport) (Rdkafka::RdkafkaError)` and returns false and we could use that for the health-check, wdyt?

Reply:
So we catch errors in the top level health check method which basically does the same thing https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_server.rb#L632, or is it better to catch here?

Reply:
also in manageiq-messaging-ready https://github.com/ManageIQ/manageiq-appliance/blob/master/COPY/usr/bin/manageiq-messaging-ready#L35

Reply:
> So we catch errors in the top level health check method which basically does the same thing

Yeah that's fine for now, I don't like exposing "internal" exception classes like `RdkafkaError` to the caller but we already do that elsewhere in this class.  Ideally we'd catch RdkafkaError and wrap it in a ManageIQ::Messaging::Error class but that's a larger effort.